lib/topological_inventory/openshift/operations/core/service_catalog_client.rb

```diff
@@ -51,6 +63,29 @@ def build_payload(service_plan_name, service_offering_name, order_parameters)
             }
           end
 
+          def wait_for_provision_complete(connection, service_instance)
+            loop do
+              sleep(sleep_poll)
+
+              service_instance = connection.get_service_instance(
+                service_instance.metadata.name,
+                service_instance.metadata.namespace
+              )
+
+              condition = service_instance.status.conditions.first
+              logger.info("#{service_instance.metadata.name}: message [#{condition.message}] status [#{condition.status}] reason [#{condition.reason}]")
+
+              break unless service_instance_provisioning?(service_instance)
+            end
+
+            service_instance
+          end
+
+          def service_instance_provisioning?(service_instance)
+            reason = service_instance.status.conditions.first&.reason
+            %w(Provisioning ProvisionRequestInFlight).include?(reason)
```

Comment:
@mkanoor I got these from a script you sent me a while ago, but I'm also seeing `OrphanMitigationComplete` have you seen this state?  It seems to flip back and forth between that and `ProvisionRequestInFlight`

Reply:
Might be related to https://github.com/kubernetes-incubator/service-catalog/issues/2514

Reply:
hm, might be safer to check for finished and error states?

Reply:
Yeah that's a good idea, hopefully there are fewer of those :)